lib/platformUtils/windows10Utils.js

```diff
@@ -45,54 +85,60 @@ function replaceManifestValues(w3cManifestInfo, content) {
+  var applicationContentUriRules = '';
+  var acurList = [];
 
-  // Set the base access rule using the start_url's base url
-  var baseUrlPattern = url.resolve(w3cManifest.start_url, '/');
-  var baseApiAccess = 'none';
+  // Set the base acur rule using the start_url's base url
+  baseAcurMatch = url.resolve(w3cManifest.start_url, '/');
   if (w3cManifest.scope && w3cManifest.scope.length) {
     // If the scope is defined, the base access rule is defined by the scope
     var parsedScopeUrl = url.parse(w3cManifest.scope);
 
     if (parsedScopeUrl.host && parsedScopeUrl.protocol) {
-      baseUrlPattern = w3cManifest.scope;
+      baseAcurMatch = w3cManifest.scope;
     } else {
-      baseUrlPattern = url.resolve(baseUrlPattern, w3cManifest.scope); 
+      baseAcurMatch = url.resolve(baseAcurMatch, w3cManifest.scope); 
     }
   }
   
-  // If the base access rule ends with '/*', remove the '*'.
-  if (baseUrlPattern.indexOf('/*', baseUrlPattern.length - 2) !== -1) {
-    baseUrlPattern = baseUrlPattern.substring(0, baseUrlPattern.length - 1);
+  // Add base rule to ACUR list
+  tryAddAcurToList(acurList, { 'match': baseAcurMatch, 'type': 'include' });
+
+  // Add rules from mjs_access_whitelist to ACUR list
+  // TODO: mjs_access_whitelist is deprecated. Should be removed in future versions
+  if (w3cManifest.mjs_access_whitelist) {
+    w3cManifest.mjs_access_whitelist.forEach(function(whitelistRule) {
+      tryAddAcurToList(acurList, { 'match': whitelistRule.url, 'type': 'include', 'runtimeAccess': whitelistRule.apiAccess });
+    });
   }
-
-  var applicationContentUriRules = '';
-
-  // Add additional access rules
-  if (w3cManifest.mjs_access_whitelist && w3cManifest.mjs_access_whitelist instanceof Array) {    
-    for (var j = 0; j < w3cManifest.mjs_access_whitelist.length; j++) {
-      var accessUrl = w3cManifest.mjs_access_whitelist[j].url;
-      // Ignore the '*' rule 
-      if (accessUrl !== '*') {
-        // If the access url ends with '/*', remove the '*'.
-        if (accessUrl.indexOf('/*', accessUrl.length - 2) !== -1) {
-          accessUrl = accessUrl.substring(0, accessUrl.length - 1);
-        }
-        
-        var apiAccess = w3cManifest.mjs_access_whitelist[j].apiAccess || 'none'; 
-        
-        if (accessUrl === baseUrlPattern) {
-          baseApiAccess = apiAccess;
-        } else  {                     
-          applicationContentUriRules += indentationChars + '<uap:Rule Type="include" WindowsRuntimeAccess="' + apiAccess + '" Match="' + accessUrl + '" />';
-        }
-      }
-    }
+  
+  // Add rules from mjs_extended_scope to ACUR list
+  if (w3cManifest.mjs_extended_scope) {
+    w3cManifest.mjs_extended_scope.forEach(function(whitelistRule) {
+      tryAddAcurToList(acurList, { 'match': whitelistRule.url, 'type': 'include' });
+    });
   }
+  
+  // Add rules from mjs_api_access to ACUR list
+  if (w3cManifest.mjs_api_access) {
+    w3cManifest.mjs_api_access.forEach(function (apiRule) {
```

Comment:
this one could probably be split up. The map. and indexOf inside the if statement is a little hard follow.